File: 프로그래머스/1/42748. K번째수/K번째수.swift
import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    
    var result : [Int] = []
    
    for i in commands {
        let bucket = array[i[0] - 1..<i[1]].sorted(by : <)
        result.append(bucket[i[2] - 1])
    }
    
    return result
    
}

// 1. commands 순회탐색
// 2. 탐색한 것에서 숫자 추출.
// 3. 추출한 것으로 array에서 부분 문자열 추출

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        
        for (int i = 0; i < commands.length; i++) {
            int[] cmd = commands[i];
            int startIdx = cmd[0]-1;
            int endIdx = cmd[1]-1;
            int cuttedSelectedIdx = cmd[2] - 1;
            
            // 잘라서 새로운 객체를 만들고 정렬해서 고른다.
            int[] newArr = new int[endIdx-startIdx+1];
            for (int j = 0; j < newArr.length; j++) {
                newArr[j] = array[startIdx + j];
            }
            
            Arrays.sort(newArr);
            answer[i] = newArr[cuttedSelectedIdx];
        }
        
        return answer;
    }
}